src/lazy/any_encoding.rs

```diff
@@ -0,0 +1,535 @@
+            },
+            LazyRawValueKind::Binary_1_0(v) => LazyRawAnySequence {
+                encoding: LazyRawSequenceKind::Binary_1_0(LazyRawBinarySequence::from_value(v)),
+            },
+        }
+    }
+}
+
+pub struct RawAnySequenceIterator<'data> {
+    encoding: RawAnySequenceIteratorKind<'data>,
+}
+
+pub enum RawAnySequenceIteratorKind<'data> {
+    Text_1_0(RawTextSequenceIterator<'data>),
+    Binary_1_0(RawBinarySequenceIterator<'data>),
+}
+
+impl<'data> Iterator for RawAnySequenceIterator<'data> {
+    type Item = IonResult<LazyRawAnyValue<'data>>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match &mut self.encoding {
+            RawAnySequenceIteratorKind::Text_1_0(i) => i
+                .next()
+                .map(|value_result| value_result.map(|value| value.into())),
+            RawAnySequenceIteratorKind::Binary_1_0(i) => i
+                .next()
+                .map(|value_result| value_result.map(|value| value.into())),
+        }
+    }
+}
+
+impl<'data> LazyRawSequence<'data, AnyEncoding> for LazyRawAnySequence<'data> {
+    type Iterator = RawAnySequenceIterator<'data>;
+
+    fn annotations(&self) -> <AnyEncoding as LazyDecoder<'data>>::AnnotationsIterator {
+        todo!()
+    }
+
+    fn ion_type(&self) -> IonType {
+        match &self.encoding {
+            LazyRawSequenceKind::Text_1_0(s) => s.ion_type(),
+            LazyRawSequenceKind::Binary_1_0(s) => s.ion_type(),
+        }
+    }
+
+    fn iter(&self) -> Self::Iterator {
+        match &self.encoding {
+            LazyRawSequenceKind::Text_1_0(s) => RawAnySequenceIterator {
+                encoding: RawAnySequenceIteratorKind::Text_1_0(s.iter()),
+            },
+            LazyRawSequenceKind::Binary_1_0(s) => RawAnySequenceIterator {
+                encoding: RawAnySequenceIteratorKind::Binary_1_0(s.iter()),
+            },
+        }
+    }
+
+    fn as_value(&self) -> LazyRawAnyValue<'data> {
+        match &self.encoding {
+            LazyRawSequenceKind::Text_1_0(s) => (s.as_value()).into(),
+            LazyRawSequenceKind::Binary_1_0(s) => (s.as_value()).into(),
+        }
+    }
+}
+
+impl<'data> From<LazyRawTextSequence<'data>> for LazyRawAnySequence<'data> {
+    fn from(value: LazyRawTextSequence<'data>) -> Self {
+        LazyRawAnySequence {
+            encoding: LazyRawSequenceKind::Text_1_0(value),
+        }
+    }
+}
+
+impl<'data> From<LazyRawBinarySequence<'data>> for LazyRawAnySequence<'data> {
+    fn from(value: LazyRawBinarySequence<'data>) -> Self {
+        LazyRawAnySequence {
+            encoding: LazyRawSequenceKind::Binary_1_0(value),
+        }
+    }
+}
+
+// ===== Structs =====
+
+#[derive(Debug, Clone)]
+pub struct LazyRawAnyStruct<'data> {
+    encoding: LazyRawStructKind<'data>,
+}
+
+#[derive(Debug, Clone)]
+pub enum LazyRawStructKind<'data> {
+    Text_1_0(LazyRawTextStruct<'data>),
+    Binary_1_0(LazyRawBinaryStruct<'data>),
+}
```

Comment:
Are these `LazyRawAny*` structs exposed to users of the crate, or are they just `pub` in this module? If just `pub` in this module, is there any reason these can't be collapsed to remove the extra layer of indirection? (And similarly for the other struct/enum combos.)

```suggestion
#[derive(Debug, Clone)]
pub struct LazyRawAnyStruct<'data> {
    Text_1_0(LazyRawTextStruct<'data>),
    Binary_1_0(LazyRawBinaryStruct<'data>),
}
```

Reply:
Offline discussion—the structs will eventually be public.

Reply:
The plan is for the `struct`s to be `pub`, but the enums to be hidden. That way they don't have to be marked `non_exhaustive` and we can add other formats/versions over time.